#!/usr/bin/env python
PACKAGE = "robot_path_costmap"

import sys
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("enabled", 		bool_t, 	0, "Whether to apply this plugin or not ", 									True)

gen.add("filter_strength", 	double_t, 		0, "Defines how much of the calculated kernel value is added to the costmap", 					1,  0, 10)
gen.add("filter_size", 		int_t, 		0, "Defines the size of the kernel. Resolution: 0.050000 meters / pixel. 4 pixel needed for robot itself", 	9,  4, 25)

gen.add("side_inflation", 	bool_t, 	0, "Whether to apply the side inflation or not", 								False)
gen.add("inflation_strength", 	double_t, 		0, "Defines how much the side should be inflated, too, compared to normal filter strength.", 			0.05,  0, 1)

gen.add("gauss_sigma", 		double_t, 	0, "Defines sigma of the gauss kernel", 									1, 0, 1000)
gen.add("gauss_s", 		double_t, 	0, "Defines 'var' of's' of the gauss kernel formula (s=var*sigma^2).", 						2,  0, 1000)

exit(gen.generate(PACKAGE, "robot_path_costmap", "NavigationPathLayer"))
